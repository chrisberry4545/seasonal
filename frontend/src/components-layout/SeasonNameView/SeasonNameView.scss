.c-season-name-view {
  display: block;
  text-align: center;
}
