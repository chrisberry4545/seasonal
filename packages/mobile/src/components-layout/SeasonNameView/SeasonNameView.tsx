import React, { FC } from 'react';

import {
  ISeasonNameViewInputProps
} from './SeasonNameView.interface';
import { TextStyle } from 'react-native';
import { TextHeadingMedium } from '../../components-elements';

const styleSeasonNameView: TextStyle = {
  marginTop: 8,
  textAlign: 'center'
};

export const SeasonNameView: FC<ISeasonNameViewInputProps> = ({
  name
}) => (
  <TextHeadingMedium style={styleSeasonNameView}>
    {name}
  </TextHeadingMedium>
);
