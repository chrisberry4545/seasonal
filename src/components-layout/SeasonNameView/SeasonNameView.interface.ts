export interface ISeasonNameViewProps {
  name: string | undefined;
}
