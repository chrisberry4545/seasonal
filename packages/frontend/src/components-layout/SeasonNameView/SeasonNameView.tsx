import React, { FC } from 'react';
import './SeasonNameView.scss';
import {
  ISeasonNameViewProps
} from './SeasonNameView.interface';
import { TextHeadingSmall } from '@chrisb-dev/seasonal-shared-frontend-components';

export const SeasonNameView: FC<ISeasonNameViewProps> = ({
  name
}) => (
  <TextHeadingSmall className='c-season-name-view'>
    {name}
  </TextHeadingSmall>
);
