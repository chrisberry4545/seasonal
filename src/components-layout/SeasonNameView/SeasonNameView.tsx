import React, { FC } from 'react';
import './SeasonNameView.scss';
import {
  ISeasonNameViewProps
} from './SeasonNameView.interface';
import { TextHeadingSmall } from '../../components-elements';

export const SeasonNameView: FC<ISeasonNameViewProps> = ({
  name
}) => (
  <TextHeadingSmall className='c-season-name-view'>
    {name}
  </TextHeadingSmall>
);
