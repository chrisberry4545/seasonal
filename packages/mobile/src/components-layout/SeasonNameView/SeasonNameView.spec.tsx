import React from 'react';
import { SeasonNameView } from './SeasonNameView';
import { shallow, ShallowWrapper } from 'enzyme';

jest.mock('../../components-elements', () => ({
  TextHeadingMedium: () => 'TextHeadingMedium'
}));

describe('<SeasonNameView />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() =>
    wrapper = shallow(
      <SeasonNameView name='Season name' />
    )
  );

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());
});
