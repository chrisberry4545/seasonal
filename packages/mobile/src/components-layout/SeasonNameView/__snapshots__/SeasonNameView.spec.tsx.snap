// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonNameView /> renders correctly 1`] = `
<TextHeadingMedium
  style={
    Object {
      "marginTop": 8,
      "textAlign": "center",
    }
  }
>
  Season name
</TextHeadingMedium>
`;
